lib/state_machine/state_machine.cpp

```diff
@@ -0,0 +1,34 @@
+#include "state_machine.hpp"
+
+namespace hyped::state_machine {
+
+// TODOLater implement
+std::optional<Message> StateMachine::handleData()
```

Comment:
Should this not be `checkTransiton` or similar?

Reply:
The name just doesn't seem to describe the functionality of this?

Reply:
https://github.com/Hyp-ed/hyped-2023/pull/69#discussion_r1083266641

Reply:
I prefer `checkTransition` to `handleData` tbh

Reply:
@mifrandir what was the problem with `checkTransition`?